emote/nn/layers.py

```diff
@@ -174,3 +175,159 @@ def get_encoder_output_size(self):
             out_size = np.prod(out_size)
 
         return out_size
+
+
+class FullyConnectedEncoder(nn.Module):
+    def __init__(
+        self,
+        input_size: int,
+        output_size: int,
+        device: torch.device,
+        condition_size: int = 0,
+        hidden_sizes: list[int] = None,
+        freeze_bn: bool = True,
+    ):
+        super().__init__()
+        self.device = device
+
+        self.input_size = input_size
+        self.output_size = output_size
+        self._freeze_bn = freeze_bn
+
+        num_layers = len(hidden_sizes)
+        batch_norm = (
+            FrozenBatchNorm1d(hidden_sizes[0])
```

Comment:
Does the frozen BN need any extra code to restore correctly from checkpoints?

Reply:
I believe It will inherit it from the _NormBase class. The only difference to the base class is that it ensures no further updates in the forward pass.